### PR TITLE
add more exception types to distinguish CHERI page fault types

### PR DESCRIPTION
**Pull request by tariqkurd-repo**
Fixes https://github.com/riscv/riscv-cheri/issues/470

### CODE REVIEW

File: src/riscv-integration.adoc [hunk 1]
```diff
@@ -801,18 +801,23 @@ xref:mtval2-cheri-causes[xrefstyle=short] respectively.
...
 |==============================================================================
 
+^1^ if {cheri_pte_ext_name} is implemented
+
 .Encoding of CAUSE field
 [#mtval2-cheri-causes,width=55%,float="center",align="center",options=header]
 |==============================================================================
 | CHERI Cause Code              | Description
-| {cheri_excep_cause_tag}       | Tag violation
+| {cheri_excep_cause_tag}       | Tag violation, or TYPE is a <<cheri_pte_ext,PTE>> fault
```
Comment (jrtc27) [inline src/riscv-integration.adoc:820]: Oh no, you don't want to lump this under CHERI faults. It's still a page fault, just with extra information.

Comment (tariqkurd-repo) [reply src/riscv-integration.adoc:820]: this just means that the field isn't populated for the PTE faults

Comment (tariqkurd-repo) [reply src/riscv-integration.adoc:820]: the TYPE field is all you need, and CAUSE=0 if it's a PTE fault, so it's redundant

Comment (tariqkurd-repo) [reply src/riscv-integration.adoc]: reworked

**Pull request merged**